iothub_client/src/iothubtransport_amqp_common.c

```diff
@@ -495,7 +495,8 @@ static int subscribe_methods(AMQP_TRANSPORT_DEVICE_INSTANCE* deviceState)
 
         if ((amqp_connection_get_session_handle(deviceState->transport_instance->amqp_connection, &session_handle)) != RESULT_OK)
         {
-            LogError("Device '%s' failed subscribing for methods (failed getting session handle)", STRING_c_str(deviceState->device_id));
+            const char* device_id = STRING_c_str(deviceState->device_id); // advoid MU_P_OR_NULL double call
```

Comment:
```suggestion
            const char* device_id = STRING_c_str(deviceState->device_id); // avoid MU_P_OR_NULL double call
```
nit: all instances of this